ecl/hql/hqlgram.y

```diff
@@ -4776,8 +4776,24 @@ fieldAttr
                         {
                             $$.setExpr(createExprAttribute(xmlDefaultAtom, $3.getExpr()), $1);
                         }
-    | DEFAULT '(' constExpression ')'
+    | DEFAULT '(' goodObject ')'
                         {
+                            IHqlExpression * arg = $3.queryExpr();
+                            if (arg->getOperator() == no_temptable && arg->queryChild(0)->queryType()==nullptr)
+                            {
+                                $3.release().setExpr(createValue(no_null));
+                            }
+                            else
+                            {
+                                ITypeInfo * type = $3.queryExprType();
+                                if (!type || type->getTypeCode() == type_set)
```

Comment:
Does this test add anything to the test in the following lines?  You could use the isNullList() function.